engine/table/src/main/java/io/deephaven/engine/table/impl/QueryTable.java

```diff
@@ -1162,20 +1160,18 @@ private void completeRefilterUpdate(
             update.added = newMapping.minus(getRowSet());
             final WritableRowSet postShiftRemovals = getRowSet().minus(newMapping);
 
-            // Update our index in post-shift keyspace.
-            getRowSet().writableCast().remove(postShiftRemovals);
-            getRowSet().writableCast().insert(update.added);
+            getRowSet().writableCast().resetTo(newMapping);
 
             // Note that removed must be propagated to listeners in pre-shift keyspace.
             if (upstream != null) {
                 upstream.shifted().unapply(postShiftRemovals);
             }
-            update.removed.writableCast().insert(postShiftRemovals);
+            update.removed = postShiftRemovals;
```

Comment:
I think you should close `removed`, or just never assign it.

Reply:
There was a comment related to this in the other part of the code where removed is created originally.  Because the shift aware updates support added + removed, we need to include the original removed from the update, plus our new removals to account for rows that are added/removed as a modify w/o lighting up the entire MCS.

Reply:
I buy that argument. I'm not sure that add-processing is consistent with it.
```
            update.added = newMapping.minus(getRowSet());
```